shared/featureconfig/flags.go

```diff
@@ -122,6 +122,10 @@ var (
 		Name:  "enable-optimized-balance-update",
 		Usage: "Enables the optimized method of updating validator balances.",
 	}
+	enableDoppleGangerProtection = &cli.BoolFlag{
+		Name:  "enable-doppleganger",
+		Usage: "Enables the validator to perform a doppleganger check.",
```

Comment:
Please add some kind of warning. This is not a 100% effective check and the user should not rely on this feature to allow them to run in unsafe configurations.